src/ledger/backend/tokencache.ts

```diff
@@ -188,7 +192,7 @@ export class TokenFetcher implements TokenProvider {
 		ttype: TokenType,
 		token: string | Address,
 	): Responder {
-		token = token instanceof Address ? token.toString() : token;
+		token = token instanceof Address ? token.key : token.toLowerCase();
```

Comment:
This is the same code as the `private ensureAddrKey` in the other commit. Maybe make `ensureAddrKey` an exported function in the same package as `Address`? Maybe also rename it to `addressKey` then ^^ And maybe use it instead of "manual" `addr.toLowerCase()` calls. If we use `addressKey` function everywhere, should we ever want to change the key coding (which we hopefully don't), we don't need to search the whole codebase for `toLowerCase()` calls.